src/main/java/by/exadel/internship/service/impl/UserTimeSlotServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+        userTimeSlotWithUserDTO.setUser(userDTO);
+        userTimeSlotWithUserDTO.setStartDate(time.getStartDate());
+        switch (userDTO.getInterviewTime()) {
+            case HALF_HOUR: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(30));
+                break;
+            }
+            case HOUR: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(60));
+                break;
+            }
+            case HOUR_HALF: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(90));
+                break;
+            }
+        }
+        userTimeSlotRepository.save(mapper.toTimeForCallUserEntity(userTimeSlotWithUserDTO));
+        log.info("User time was restoring");
+    }
+
+    private void checkTime(UserTimeSlotDTO time) {
+        log.info("Bringing time to a common form");
+        time.setStartDate(commonTimeForm(time.getStartDate()));
+        time.setEndDate(commonTimeForm(time.getEndDate()));
+        log.info("Time brought to a common form");
+    }
+
+    private LocalDateTime commonTimeForm(LocalDateTime dateTime) {
+        if (dateTime.getMinute() % 60 < 30) {
+            return  LocalDateTime.of(dateTime.getYear(), dateTime.getMonth(),
+                    dateTime.getDayOfMonth(), dateTime.getHour(),
+                    DEFAULT_START_MINUTES_IF_BETWEEN_ZERO_THIRTY);
+        }
+        if (dateTime.getMinute() % 60 > 30){
+            return  LocalDateTime.of(dateTime.getYear(), dateTime.getMonth(),
+                    dateTime.getDayOfMonth(), dateTime.getHour() + 1,
+                    DEFAULT_START_MINUTES_IF_BETWEEN_THIRTY_ZERO);
+        }
+        throw new RuntimeException("Time exception can't to do common form for time");
+    }
+
+    private void separateTime(UserTimeSlotDTO time, InterviewTime interviewTimeUser, List<UserTimeSlotWithUserDTO> resultUSerTimeList) {
+        log.info("Separate time on part");
+        List<UserTimeSlotWithUserDTO> newUserTimeList = new ArrayList<>();
+        Duration duration = Duration.between(time.getStartDate(), time.getEndDate());
+        long numberOfPeriods = determineTime(duration, interviewTimeUser);
+        for (int i = 0; i < numberOfPeriods; i++) {
+            UserTimeSlotWithUserDTO tempUserTime = new UserTimeSlotWithUserDTO();
+            tempUserTime.setStartDate(time.getStartDate());
+            tempUserTime.setEndDate(tempUserTime.getStartDate().plusMinutes(INTERVIEW_TIME));
+            newUserTimeList.add(tempUserTime);
+            time.setStartDate(tempUserTime.getEndDate());
+        }
+        resultUSerTimeList.addAll(newUserTimeList);
+        log.info("Time separated on part");
+    }
+
+    private long determineTime(Duration duration, InterviewTime interviewTimeUser) {
+        log.info("Return of temporary interview");
+        switch (interviewTimeUser) {
+            case HALF_HOUR: {
+                INTERVIEW_TIME = 30;
```

Comment:
every request will change INTERVIEW_TIME on class level. It's not thread-safe. Use local variable or even better add field minutes to enum `InterviewTime`

Reply:
Done